datasetinsights/datasets/transformers/coco.py

```diff
@@ -0,0 +1,144 @@
+import hashlib
+import json
+import logging
+import shutil
+import uuid
+from pathlib import Path
+
+from PIL import Image
+
+import datasetinsights.constants as const
+from datasetinsights.datasets.unity_perception import (
+    AnnotationDefinitions,
+    Captures,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def uuid_to_int(input_uuid):
+    try:
+        u = uuid.UUID(input_uuid).int
+    except (AttributeError, ValueError):
+        u = int(
+            hashlib.md5(str(input_uuid).encode("utf8")).hexdigest(), base=16
+        )
+
+    return u
+
+
+class COCOTransformer:
+    """Convert Synthetic dataset to COCO format.
+
+    Args:
+        data_root (str): root directory of the dataset
+        bbox2d_definition_id (str): the annotation definition id for
+            2d bounding boxes in this dataset.
+    """
+
+    def __init__(self, data_root, bbox2d_definition_id):
+        self._data_root = Path(data_root)
+        self._bbox2d_definition_id = bbox2d_definition_id
+        self._captures = Captures(
+            data_root=data_root, version=const.DEFAULT_PERCEPTION_VERSION
+        ).filter(def_id=bbox2d_definition_id)
+        self._annotation_definitions = AnnotationDefinitions(
+            data_root, version=const.DEFAULT_PERCEPTION_VERSION
+        )
+
+    def execute(self, output):
+        """Execute COCO Transformer
+
+        Args:
+            output (str): the output directory where converted dataset will
+              be stored.
+        """
+        self._copy_images(output)
+        self._process_instances(output)
+
+    def _copy_images(self, output):
+        image_to_folder = Path(output) / "images"
+        image_to_folder.mkdir(parents=True, exist_ok=True)
+        for _, row in self._captures.iterrows():
+            image_from = self._data_root / row["filename"]
+            if not image_from.exists():
+                continue
+            capture_id = uuid_to_int(row["id"])
+            image_to = image_to_folder / f"camera_{capture_id}.png"
```

Comment:
@salehe-ee says: I think maintaining the RGB filename to be the same as the original file would be better. Currently the UUID is converted to int and the filename is written as f"camera_{capture_id}.png"

Reply:
This is actually intentional to mimic the structure in the real world COCO dataset. For example, the following record found in `instances_val2017.json` file indicates that the image has "id" field `174482` and the filename is `000000174482.jpg`. 

```json
{"license": 6,"file_name": "000000174482.jpg","coco_url": "http://images.cocodataset.org/val2017/000000174482.jpg","height": 388,"width": 640,"date_captured": "2013-11-14 23:16:55","flickr_url": "http://farm8.staticflickr.com/7020/6478877255_242f741dd1_z.jpg","id": 174482}
```